graph/src/blockchain/polling_block_stream.rs

```diff
@@ -531,11 +544,35 @@ impl<C: Blockchain> Stream for PollingBlockStream<C> {
                             // Reset error count
                             self.consecutive_err_count = 0;
 
-                            // Switch to idle
-                            self.state = BlockStreamState::Idle;
-
-                            // Poll for chain head update
-                            continue;
+                            // Switch to idle. When idle, wait for a chain head update notification,
+                            // or as a fallback poll for an update after a timeout.
+                            let mut watcher = self.chain_head_update_watcher.clone();
+                            let watcher_fut: Pin<Box<dyn Future<Output = _> + Send>> = async move {
+                                watcher
+                                    .changed()
```

Comment:
On further testing, because this watcher will always have an unseen change since the original `chain_head_update_watcher` wasn't checked for the change. This is a deadly flaw in the PR since I don't see a good way to solve this, I'll rework this fallback with a different approach.